core/src/main/java/org/polypheny/db/algebra/core/document/DocumentFilter.java

```diff
@@ -49,7 +49,9 @@ protected DocumentFilter( AlgOptCluster cluster, AlgTraitSet traits, AlgNode inp
 
     @Override
     public String algCompareString() {
-        return "$" + getClass().getSimpleName() + "$" + condition.hashCode() + "$" + input.algCompareString();
+        return getClass().getSimpleName() + "$"
+                + condition.hashCode() + "$"
```

Comment:
Is the hash code stable enough for the purposes of this compare sting?

Reply:
Yes, as it uses the same approach as in the relational Filter, but does not allow for a null as filter. (input algCompString is not visible in this comment but also belongs to this.)